src/events.rs

```diff
@@ -109,6 +109,25 @@ where E: Send + 'static
             }
         }
     }
+
+    /// A non-blocking receive which returns instantly.
+    /// Returns Some(E) if an event was received by this queue, otherwise returns None.
+    pub fn try_recv(&mut self) -> Option<E> {
```

Comment:
I would call `try_receive()` in order to maintain the parallelism with the receive family functions